aten/src/ATen/templates/Type.h

```diff
@@ -51,6 +51,7 @@ struct AT_API Type {
 
   virtual ~Type() {}
   virtual ScalarType scalarType() const = 0;
+  virtual caffe2::TypeMeta typeMeta() const = 0;
```

Comment:
Consider `const caffe2::TypeMeta&` instead, since TypeMeta is fat.

Reply:
TypeMeta is fine to use, I'm about to land something that makes TypeMeta lean.